packages/plugin-ext/src/plugin/window-state.ts

```diff
@@ -50,16 +50,20 @@ export class WindowStateExtImpl implements WindowStateExt {
     }
 
     openUri(uri: URI): Promise<boolean> {
+        if (!uri.scheme.trim().length) {
+            throw new Error('Invalid scheme - cannot be empty');
+        }
+
+        if (Schemes.http !== uri.scheme && Schemes.https !== uri.scheme) {
+            throw new Error(`Invalid scheme '${uri.scheme}'`);
+        }
         return this.proxy.$openUri(uri);
     }
 
     async asExternalUri(target: URI): Promise<URI> {
         if (!target.scheme.trim().length) {
             throw new Error('Invalid scheme - cannot be empty');
         }
-        if (Schemes.http !== target.scheme && Schemes.https !== target.scheme) {
```

Comment:
If I read the documentation at https://code.visualstudio.com/api/references/vscode-api#env correctly, `asExternalUri` treats `http`, `https` and the value of `vscode.env.uriScheme` (I suspect it's `theia`) specially and converts anything else into a workspace uri. I don't think that is what's happening on the main side. Removing the check is correct, IMO, though.